guides/cookbook.md

```diff
@@ -725,6 +804,52 @@ class PrettyPrint
 ```
 </div>
 
+<div class="tabpane Go" markdown="1">
+Ion data can be pretty-printed using `NewTextWriterOpts` and pass `TextWriterPretty` option to it:
+```Go
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/amzn/ion-go/ion"
+)
+
+type l3 struct {
+	Level3 string `ion:"level3"`
+}
+type l2 struct {
+	Level2 l3 `ion:"level2"`
+	X      int `ion:"x"`
+}
+type l1 struct {
+	Level1 l2 `ion:"level1"`
+	Y      []string `ion:"y"`
+}
+
+func main() {
+	l3Val := l3{"foo"}
+	l2Val := l2{l3Val, 2}
+	l1Val := l1{l2Val, []string{"a", "b", "c"}}
+
+	buf := strings.Builder{}
+	writer := ion.NewTextWriterOpts(&buf, ion.TextWriterPretty)
+
+	err := ion.MarshalTo(writer, l1Val)
+	if err != nil {
+		panic(err)
+	}
+
+	if err := writer.Finish(); err != nil {
+		panic(err)
+	}
+
+	fmt.Println(buf.String())
```

Comment:
Can you comment what this prints

Reply:
This is to show the pretty print feature and the output is going to be:
```
   level1: {
     level2: {
       level3: "foo"
     },
     x: 2
   },
   y: [
     a,
     b,
     c
   ]
 }
```
so I cannot put the output as an inline comment.
Also the section where this code is placed, shows the expected output of pretty print